src/object/srv_obj.c

```diff
@@ -1110,10 +1179,50 @@ ds_iter_vos(crt_rpc_t *rpc, struct vos_iter_anchors *anchors,
 	 * Need to use separate anchors for SV and EV, or return a
 	 * 'type' to indicate the anchor is on SV tree or EV tree.
 	 */
-	if (type == VOS_ITER_SINGLE)
+	if (type == VOS_ITER_SINGLE) {
 		anchors->ia_sv = anchors->ia_ev;
+	} else if (type == VOS_ITER_DKEY &&
+		   (daos_anchor_get_flags(&anchors->ia_dkey) &
+		    DIOF_FLUSH_DTX)) {
```

Comment:
I think we discussed about this and agreed we can force commit all outstanding DTXs for the specified object right?
Seems you are committing all DTXs at here? If this is just for the first step implementation, I'm totally OK, but please add comments for it if this is true.